### ISSUE TITLE
SocketException: Connection reset while syncing with server on connect

### ISSUE THREAD

**Issue by huntingtiger1004** [sleepingdragoninn/antique-atlas#100]
Okay, so I am genuinely perplexed by this issue, so I'm not quite sure what to provide or where to look to be more helpful in diagnosing this issue. If I weren't, I'd have more information I could provide out the gate.

I've been co-hosting a server with surveyor and antique atlas in the modpack that I put together (you might know me from issue #98 posted by spencer012 last week). We hadn't encountered any additional weird behavior anywhere right up until we updated the modpack to include more mods. This issue persists across launchers (modrinth, and ATLauncher are the only two I'm certain of right now), and is as follows:

For whatever reason, after the pack is updated we are prevented from joining the server. It usually just spits out an "Internal Exception: java.net.SocketException: connection reset" error the frame that we load into the world. Now I was under the assumption that this issue was due to some other mod or another that was changed in the update, however, upon someone's removal of antique atlas/surveyor, they could join fine (since that was a previous temporary solution). Now this struck me as strange, so I downloaded a fresh instance of the mod and transferred the "data" folder over... and it worked perfectly fine?

Rejoining the server using the broken instance is still blocked, but the new instance still joins fine with all map data properly detected. These join issues are not present on the primary version I use to export the pack update when I was testing it out. I'm not including any extra data beyond the config folder and the mods folder (at least that's seemingly relevant. I do also include a shaderpacks and datapacks folder). Duplicating the profile on modrinth also fixes the issue.

As of yet, I have not attempted to disable the mods on my own broken profile (on another computer) to see what is causing the issue, as I'm unsure if that will prove to be a permanent fix that makes this error unreproducable. I have been looking through any errors sent my way including my own error logs and have not seen much of anything that points towards it being a surveyor issue, but all behavior described and tested on my end has so far been occurring consistently.

As of right now, simply having people reinstall the pack and transfer their map data has been a functional temporary fix, but beyond that I am completely confused. I apologize for the lack of information in advance, but I'm not sure what would be useful data to provide that wouldn't be just pure useless bloat information. I'm unsure if there is anything that can be done on your end to fix this issue, or if it's even relevant enough to be put into the issues. Sorry overall for my cluelessness on how to conduct reporting this.

**Comment by nishant1151** [comment]
Possible Causes:

Mod Conflict: An incompatibility between the new mods and "antique atlas" or "surveyor" could be causing the connection issue.
Corrupted Mod Installation: The broken instance of the modpack might have a corrupted file within the "antique atlas" or "surveyor" folders.
Server Configuration: There might be a setting in the server configuration that conflicts with the updated mods.
Next Steps:

Isolating the Issue: Try disabling all the newly added mods on your broken profile and see if you can join the server. This will help narrow down if the problem lies with one of the new mods.
Server Log Check: Review the server logs for any error messages related to the connection issue or the newly added mods. This might provide additional clues.
Community Resources: Search online forums and communities dedicated to modded Minecraft for similar issues. You might find solutions or workarounds from other users who faced similar problems with specific mod combinations.
Report the Issue: If you can isolate the problem to a specific mod or combination of mods, consider reporting the issue to the respective mod developers.
Information to Provide (if applicable):

List of all mods in the pack, especially the newly added ones.
Specific version of Minecraft you're using.
Any relevant server configuration settings.
Error messages from the server logs (if any).
By following these steps and gathering more information, you should be able to pinpoint the cause of the connection error and find a solution. Remember, the Minecraft modding community is vast and helpful, so don't hesitate to seek assistance from online resources.

Here are some additional points to consider:

Reinstalling the entire modpack might seem tedious, but it can sometimes resolve unexpected issues due to corrupted installations.
Duplicating the profile is a good temporary solution, but it doesn't address the root cause of the problem.

**Comment by sleepingdragoninn** [comment]
Feel free to post client and server logs of the connection reset happening to mclogs

**Comment by huntingtiger1004** [comment]
Sorry for the lengthy message, I was stressing out way too much trying to figure out the issue for my friends for 4-5 hours or so, and was really trying to find more solid information that'd actually help and not just show up clueless. I did my best to look up forums and references and unfortunately (likely due to lack of experience/familiarity on my part) came up short.

We are running on Minecraft version 1.20.1 with Fabric Loader 0.15.10.

Antique atlas is still on 2.7.2 and Surveyor is on 0.4.3, as I didn't want to update any mods from the previous version that weren't reporting any fixes in the changelogs, as I primarily wanted to maintain stability with the current features that weren't causing issues. Antique Atlas and Surveyor had not been causing any issues at the time, and I didn't want to run the risk of old map data conflicting with new map data.

I have a client log, but I cannot provide a server log for the time being (but will try to do so tomorrow). I should note the server is only detecting it as a client disconnect and not really reporting any errors on the server side. We haven't altered any config relating to surveyor or antique atlas.

Broken Client Log: https://mclo.gs/YqeV8XX

Okay so, the list may be a little absurdly long to post in text format, but here's the problematic update version in question, and you can see all the mods listed there:
https://modrinth.com/modpack/stripes-journey-vibes/version/0.3.3a
There's primarily structure mods which have been causing conflicts in the past, but things that notably change world generation include:
- Create
- Hephaestus (a Fabric port of tinker's construct)
- Immersive Weathering
- BetterEnd (though I should note we've yet to visit the end)
- Farmer's delight

The only mods I can think of that actually adds new biomes are BetterEnd (all biomes successfully find fallbacks), sniffer+ (which successfully locates a fallback biome) and spawn (which fails to find a fallback biome).

Mods introduced in the new version include:
- Alternate Origins GUI (a GUI mod for the Origins Mod)
- Aileron (a mod that rebalances the elytra)
- Polymorph (a mod that lets you switch between conflicting crafting recipes)

Any errors I've been seeing are no different than previous errors we've encountered but still let the game run fine after initial popup. Once again, I'd like to reiterate that re-including the "data/surveyor" folder on a fresh install, and duplicating the pack through the launcher both cause the issue to confusingly disappear. I can also report that disabling surveyor and antique atlas on the broken instance solves the issue, and then re-enabling it afterwards and trying again has also, confusingly, fixed the issue.

So as far as I can tell, this all ends up being "did you try turning it on and off again" but just with a switch for two mods on a launcher.

Once again, sorry for coming here with some hyper-specific issue, I am no closer to understanding this, sorry for the headache.

**Comment by huntingtiger1004** [comment]
I'd also like to say I've been trying to keep a very thorough log of changes to the best of my abilities, down to config alterations and default keybind adjustments, but the majority of updates come from before the pack was even posted to modrinth and I was assembling it for my own personal usage and was content knowing that the pack I assembled would probably be extremely buggy due to my lack of experience and knowledge in the nitty-gritty of Minecraft modding and modpacks.

That said, if you would like any of the few old changelogs that are not currently posted anywhere online I could compile information to a pastebin or something for you to access on the off-chance it might assist the process of troubleshooting.

**Comment by sleepingdragoninn** [comment]
client log indicates the issue is on the server - can I get a server log during a period where this failed connect occurs?

**Comment by huntingtiger1004** [comment]
Here's a log from the server, but it is only for the specific time frame, as the server's uptime has been going for a long time now, and I'm currently waiting for a lull activity but everyone appears to be having a late night...
https://mclo.gs/AT1q28L
Apparently my friend who's running the machine didn't configure the timezones properly and I only just noticed, so excuse the time differences.

As I've been asking around more, it turns out that strangely some people encountered the problem, but others didn't. On average though, anyone who couldn't connect simply had to do something with surveyor/antique atlas on their side. Unfortunately, I lack a lot of outside data because most people fixed the problem themselves before I learned that it was related to the map and moved on without saying anything.

Once things slow down I'll make a copy of the server so I can have a proper test environment and provide a full log (assuming that would make any sense. If anything I'll provide a quieter log on the main server first when I get the chance.)

**Comment by huntingtiger1004** [comment]
Actually having written that- is there any chance that the time zones being wrong might have anything to do with it?

**Comment by sleepingdragoninn** [comment]
There's definitely a timeout happening here, and it's probably surveyor timing out while attempting to perform a specific type of sync action - but honestly I'm unclear what and unsure if there's a useful utility for timeouts (maybe login profiling the client?) 

On the client, surveyor only saves state to the .minecraft/data folder and (technically) the config/surveyor.toml file. If those are preserved and the mod version is the same, I'd expect to see the same presence of any issues.

**Comment by sleepingdragoninn** [comment]
For the record, I've known bclib (betterend) to break things you really wouldn't expect it to - if it really is a mod messing with surveyor and not just a surveyor bug (doubt), that's my first guess.

**Comment by huntingtiger1004** [comment]
After noticing the time zone detail I'm personally suspicious of the time mismatch creating some kind of confusion in communication resulting in it attempting to overwrite basically the entire map unnecessarily. Would explain why it becomes a major issue the longer someone's been active on the server, and why some people may not encounter it if they have a playstyle that leads to them being mostly stationary. I also feel it relevant to mention that after doing some more fiddling with the broken instance after turning the mods on and off again for some reason seemingly fixed it, I started encountering the same error again.

 Though that's just me spitballing as that's similar to an issue I've encountered secondhand outside of Minecraft.

That said, I am down to try whatever and will just yoink better end out of the pack on the test server since we haven't even gone to the end yet if that's what it takes to narrow things down.

**Comment by huntingtiger1004** [comment]
Er- to be clear when I make that activity comment I'm failing to speak clearly about our previous experiences clearing map data proving a temporary fix on the previous issue and less-so on having actual numbers to go off of in terms of knowing for sure it only affects those with long playtime, but again that's just me spitballing.

**Comment by huntingtiger1004** [comment]
Okay so due to an overabundance of personal life chaos this obviously has taken way longer than it should have to get a server log, and unfortunately, it's gonna take longer still.

That said, in the process of experimenting around, I've learned that upon removing the mods and their respective data from a separate version of the server, I still time out in the same way.

I can move my structures,dat and landmarks.dat files from each dimension over into a new profile and everything is fine. However, including the full map data along with it prevents me from joining. I'd also like to mention that some of the individual data files in my data/surveyor/<id>/overworld/minecraft folder range up to 973 KB ( which is c.-2.-2).

I figured I'd mention the file size specifically as I don't think that has previously come up. I'd also like to again bring attention to the spawn:ant_gardens biome failing to find a fallback due to it not being in any conventional or vanilla tag, according to the error.

Also it did just occur to me that we are using fabric seasons, and that may be confusing the map due to the switching of biome properties and whatnot.

I dunno, there's a lot going on, I don't know if any of this new information helps, I'll try to get that server log ASAP.
